plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2439,8 +2466,7 @@ void producer_plugin_impl::produce_block() {
 
    block_state_ptr new_bs = chain.head_block_state();
 
-   _account_fails.report();
-   _account_fails.clear();
+   _account_fails.report(new_bs->block_num);
```

Comment:
I think you should make it `report_and_clear(uint32_t block_num)` and only call it from start_block. The `-1` in `report` vs `clear` seems a bit fragile.